pkg/scheduling/v2/scheduler.go

```diff
@@ -518,6 +518,11 @@ func (s *Scheduler) tryAssignBatch(
 		// if the action is not in the map, then we have no slots to assign to
 		for i := range res {
 			res[i].noSlots = true
+
+			// nack the rate limit result, if it exists
+			if res[i].rateLimitResult != nil {
+				res[i].rateLimitResult.nack()
```

Comment:
This also fixes an issue where a rate limit is used when there are no slots available.